examples/aws/eks-auto-mode/README.md

```diff
@@ -0,0 +1,164 @@
+  externalUrl: http://ip.address.nip.io/
+
+ingress:
+  enabled: true
+  className: alb
+```
+
+### Configuring Network Load Balancer
+
+Example of how to deploy and expose GraphDB with [Network Load Balancer](https://docs.aws.amazon.com/eks/latest/userguide/auto-configure-nlb.html) without SSL enabled:
+
+```yaml
+service:
+  enabled: true
+  ports:
+   http: 80
+  type: LoadBalancer
+  loadBalancerClass: "eks.amazonaws.com/nlb"
+  loadBalancerSourceRanges: ["0.0.0.0/0"]
+  annotations:
+   service.beta.kubernetes.io/aws-load-balancer-name: "graphdb-lb"
+   service.beta.kubernetes.io/aws-load-balancer-nlb-target-type: "ip"
+    # This ensures the load balancer is public
+   service.beta.kubernetes.io/aws-load-balancer-scheme: "internet-facing"
+   service.beta.kubernetes.io/aws-load-balancer-attributes: "load_balancing.cross_zone.enabled=true"
+```
+
+Example of how to deploy and expose GraphDB with Network Load Balancer with SSL enabled:
+
+```yaml
+service:
+  enabled: true
+  ports:
+    http: 443
+  type: LoadBalancer
+  loadBalancerClass: "eks.amazonaws.com/nlb"
+  loadBalancerSourceRanges: ["0.0.0.0/0"]
+  annotations:
+    service.beta.kubernetes.io/aws-load-balancer-name: "graphdb-lb"
+    service.beta.kubernetes.io/aws-load-balancer-nlb-target-type: "ip"
+    # This ensures the load balancer is public
+    service.beta.kubernetes.io/aws-load-balancer-scheme: "internet-facing"
+    # ARN of the ACM SSL Certificate that will be used
+    service.beta.kubernetes.io/aws-load-balancer-ssl-cert: ""
+    service.beta.kubernetes.io/aws-load-balancer-ssl-ports: "443"
+    service.beta.kubernetes.io/aws-load-balancer-ssl-negotiation-policy: "ELBSecurityPolicy-TLS13-1-2-2021-06"
+    service.beta.kubernetes.io/aws-load-balancer-attributes: "load_balancing.cross_zone.enabled=true"
+```
+
+### Configuring custom NodePool
+
+The nodes provisioned automatically by Karpenter may use arbitrary instance types, which can lead to issues such as:
+
+  - Different CPU architectures or virtualization types across nodes
+
+  - Varying CPU counts with the same memory size, resulting in inconsistent performance
+
+  - Other unpredictable hardware differences
+
+To avoid this, you can restrict provisioning to a specific instance family and CPU architecture by using a node selector.
+
+This is typically done by creating a [Node Pool](https://docs.aws.amazon.com/eks/latest/userguide/create-node-pool.html) with the desired constraints, and then
+configuring the GraphDB and Proxy pods to schedule only onto nodes provisioned from that custom pool.
+
+```yaml
+nodeSelector:
+  karpenter.sh/nodepool: my-node-pool
+proxy:
+  nodeSelector:
+    karpenter.sh/nodepool: my-node-pool
+```
+
+### Spreading pods across different nodes
+
+To ensure that pods are scheduled on different nodes and that GraphDB and GraphDB Proxy pods do not run on the same
+node, you need to configure additional podAntiAffinity rules, for example:
+
+```yaml
+affinity:
+  podAntiAffinity:
```

Comment:
Our default hard pod anti-affinity currently tracks only the nodes or the proxies. We can extend it with a toggle to consider the proxies as well. I'll submit an improvement task.